cloud/scope/powervs_cluster.go

```diff
@@ -157,3 +261,1557 @@ func (s *PowerVSClusterScope) PatchObject() error {
+		return nil
+	}
+	// create transit gateway
+	transitGatewayID, err := s.createTransitGateway()
+	if err != nil {
+		return err
+	}
+	if transitGatewayID != nil {
+		s.SetStatus(TransitGateway, infrav1beta2.ResourceReference{ID: transitGatewayID, ControllerCreated: pointer.Bool(true)})
+		return nil
+	}
+
+	// verify that the transit gateway connections are attached
+	err = s.checkTransitGatewayStatus(transitGatewayID)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// checkTransitGateway checks transit gateway exist in cloud.
+func (s *PowerVSClusterScope) checkTransitGateway() (string, error) {
+	// TODO(karthik-k-n): Support regex
+	transitGateway, err := s.TransitGatewayClient.GetTransitGatewayByName(*s.GetServiceName(TransitGateway))
+	if err != nil {
+		return "", err
+	}
+	if transitGateway == nil {
+		return "", nil
+	}
+	if err = s.checkTransitGatewayStatus(transitGateway.ID); err != nil {
+		return "", err
+	}
+	return *transitGateway.ID, nil
+}
+
+// checkTransitGatewayStatus checks transit gateway status in cloud.
+func (s *PowerVSClusterScope) checkTransitGatewayStatus(transitGatewayID *string) error {
+	transitGateway, _, err := s.TransitGatewayClient.GetTransitGateway(&tgapiv1.GetTransitGatewayOptions{
+		ID: transitGatewayID,
+	})
+	if err != nil {
+		return err
+	}
+	if transitGateway == nil {
+		return fmt.Errorf("tranist gateway is nil")
+	}
+	if *transitGateway.Status != "available" {
+		return fmt.Errorf("error tranist gateway %s not in available status, current status: %s", *transitGatewayID, *transitGateway.Status)
+	}
+
+	tgConnections, _, err := s.TransitGatewayClient.ListTransitGatewayConnections(&tgapiv1.ListTransitGatewayConnectionsOptions{
+		TransitGatewayID: transitGateway.ID,
+	})
+	if err != nil {
+		return fmt.Errorf("error listing transit gateway connections: %w", err)
+	}
+
+	if len(tgConnections.Connections) == 0 {
+		return fmt.Errorf("no connections are attched to transit gateway")
+	}
+
+	vpcCRN, err := s.fetchVPCCRN()
+	if err != nil {
+		return fmt.Errorf("error failed to fetch VPC CRN: %w", err)
+	}
+
+	pvsServiceInstanceCRN, err := s.fetchPowerVSServiceInstanceCRN()
+	if err != nil {
+		return fmt.Errorf("error failed to fetch powervs service instance CRN: %w", err)
+	}
+
+	var powerVSAttached, vpcAttached bool
+	for _, conn := range tgConnections.Connections {
+		if *conn.NetworkType == "vpc" && *conn.NetworkID == *vpcCRN {
+			if *conn.Status != "attached" {
+				return fmt.Errorf("error vpc connection not attached to transit gateway, current status: %s", *conn.Status)
+			}
+			vpcAttached = true
+		}
+		if *conn.NetworkType == "power_virtual_server" && *conn.NetworkID == *pvsServiceInstanceCRN {
+			if *conn.Status != "attached" {
+				return fmt.Errorf("error powervs connection not attached to transit gateway, current status: %s", *conn.Status)
+			}
+			powerVSAttached = true
+		}
+	}
+	if !powerVSAttached || !vpcAttached {
```

Comment:
It seems it won't reach here if any one of the case is false since you are already returning from respective case.

Reply:
yeah, The idea is given both the cases are true, we want to make sure both vpc and powervs is attached in tg connection